pvlib/ivtools.py

```diff
@@ -0,0 +1,268 @@
+        Temperature coefficient of short circuit current at STC [A/C]
+    beta_voc : float
+        Temperature coefficient of open circuit voltage at STC [V/C]
+    gamma_pmp : float
+        Temperature coefficient of power at maximum point point at STC [%/C]
+    cells_in_series : int
+        Number of cells in series
+    temp_ref : float, default 25C
+        Reference temperature condition
+
+    Returns
+    -------
+    a_ref : float
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at reference
+        conditions, in units of V.
+
+    I_L_ref : float
+        The light-generated current (or photocurrent) at reference conditions,
+        in amperes.
+
+    I_o_ref : float
+        The dark or diode reverse saturation current at reference conditions,
+        in amperes.
+
+    R_sh_ref : float
+        The shunt resistance at reference conditions, in ohms.
+
+    R_s : float
+        The series resistance at reference conditions, in ohms.
+
+    Adjust : float
+        The adjustment to the temperature coefficient for short circuit
+        current, in percent.
+
+    Raises:
+        ImportError if NREL-PySAM is not installed
+
+    References
+    ----------
+    [1] A. Dobos, "An Improved Coefficient Calculator for the California
+    Energy Commission 6 Parameter Photovoltaic Module Model", Journal of
+    Solar Energy Engineering, vol 134, 2012.
+    '''
+
+    try:
+        from PySAM import PySSC
+    except ImportError as e:
+        raise(e)
+
+    datadict = {'tech_model': '6parsolve', 'financial_model': 'none',
+                'celltype': celltype, 'Vmp': v_mp,
+                'Imp': i_mp, 'Voc': v_oc, 'Isc': i_sc, 'alpha_isc': alpha_sc,
+                'beta_voc': beta_voc, 'gamma_pmp': gamma_pmp,
+                'Nser': cells_in_series, 'Tref': temp_ref}
+
+    result = PySSC.ssc_sim_from_dict(datadict)
+    a_ref = result['a']
+    I_L_ref = result['Il']
+    I_o_ref = result['Io']
+    R_s = result['Rs']
+    R_sh_ref = result['Rsh']
+    Adjust = result['Adj']
+
+    return I_L_ref, I_o_ref, R_sh_ref, R_s, a_ref, Adjust
+
+
+def fit_sde_sandia(v, i, v_oc, i_sc, v_mp, i_mp, vlim=0.2, ilim=0.1):
+    """ Fits the single diode equation to an IV curve.
+
+    If fitting fails, returns NaN in each parameter.
+
+    Parameters
+    ----------
+    v : numeric
+        Voltage at each point on the IV curve, from 0 to v_oc
+
+    i : numeric
+        Current at each point on the IV curve, from i_sc to 0
+
+    v_oc : float
+        Open circuit voltage
+
+    i_sc : float
+        Short circuit current
+
+    v_mp : float
+        Voltage at maximum power point
+
+    i_mp : float
+        Current at maximum power point
+
+    vlim : float, default 0.2
+        defines linear portion of IV curve i.e. V <= vlim * v_oc
+
+    ilim : float, default 0.1
+        defines exponential portion of IV curve i.e. I > ilim * i_sc
```

Comment:
Model question: Is it possible to determine `ilim` from `vlim` (or vice versa)?

This would simplify the interface somewhat.

Reply:
Ah, I suppose "exponential portion" excludes the knee of the curve. However, another interpretation of "exponential portion" is where the exponential term cannot be neglected. Is the sign correct in `I > ilim * i_sc`?

Reply:
The intent is the latter, where the exponential term is numerically meaningful. However, `ilim` protects against log(negative numbers) so it's not quite the same as current level where the exponential term is meaningful.  I hope the docstring makes this more clear. Open to suggestions. I recognize that you don't have the benefit of having the paper available yet, but the function 's usage should be clear without reference to the paper.